### PR TITLE
chore: move logger to a separate package

### PR DESCRIPTION
**Pull request by pierre-lehnen-rc**

### CODE REVIEW

File: apps/meteor/server/lib/logger/startup.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
+import { setQueueLimit, logLevel, type LogLevelSetting } from '@rocket.chat/logger';
+
 import { settings } from '../../../app/settings/server';
-import type { LogLevelSetting } from './logLevel';
-import { logLevel } from './logLevel';
-import { setQueueLimit } from './logQueue';
```
Comment (sampaiodiego) [inline apps/meteor/server/lib/logger/startup.ts:4]: I wonder if the "log queue" feature should be kept on meteor app.. it is probably not relevant to other packages/apps using the logger

Comment (pierre-lehnen-rc) [reply apps/meteor/server/lib/logger/startup.ts:4]: I moved it there to ensure it runs before the main pino instance is instantiated - so Pino can detect that `process.stdout.write` has been modified.

File: packages/logger/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+	"name": "@rocket.chat/logger",
+	"version": "0.0.1",
+	"private": true,
+	"devDependencies": {
+		"@types/jest": "^27.4.1",
+		"eslint": "~8.45.0",
+		"jest": "~29.5.0",
+		"ts-jest": "~29.0.5",
+		"typescript": "~5.0.2"
```
Comment (sampaiodiego) [inline packages/logger/package.json:10]: we're using TS 5.1 elsewhere.. can you pls update to 5.1 and also check if the other versions are also matching?

File: packages/logger/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		"dev": "tsc -p tsconfig.json --watch --preserveWatchOutput"
+	},
+	"main": "./dist/index.js",
+	"typings": "./dist/index.d.ts",
+	"files": [
+		"/dist"
+	],
+	"dependencies": {
+		"@rocket.chat/emitter": "next",
+		"pino": "^8.14.1"
```
Comment (sampaiodiego) [inline packages/logger/package.json:26]: maybe a good opportunity to update to pino 8.15?

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

**Review by sampaiodiego — DISMISSED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**